provider/consul_catalog.go

```diff
@@ -133,9 +133,11 @@ func (provider *ConsulCatalog) getBackendAddress(node *api.ServiceEntry) string
 
 func (provider *ConsulCatalog) getBackendName(node *api.ServiceEntry, index int) string {
 	serviceName := node.Service.Service + "--" + node.Service.Address + "--" + strconv.Itoa(node.Service.Port)
-	if len(node.Service.Tags) > 0 {
-		serviceName += "--" + strings.Join(node.Service.Tags, "--")
+
+	for _, tag := range node.Service.Tags {
+		serviceName += "--" + strings.Replace(tag, " ", "-", -1)
 	}
+
 	serviceName = strings.Replace(serviceName, ".", "-", -1)
 	serviceName = strings.Replace(serviceName, "=", "-", -1)
```

Comment:
Running `normalize` on the result strips away the `--` and replaces that with simple `-` so I guess the best way would be to run normalise only on the tags?

or should normalize be modified to allow `--`?